github/compliance/issue.py

```diff
@@ -234,20 +236,27 @@ def create_or_update_issue(
     latest_processing_date: datetime.date|datetime.datetime=None,
     extra_labels: typing.Iterable[str]=None,
     preserve_labels_regexes: typing.Iterable[str]=(),
+    ctx_labels: typing.Iterable[str]=(),
```

Comment:
a docstr - _briefly_ explaining the semantics of non-obvious parameters (such as ctx_labels) would be nice.

in particular, there are both extra_labels and ctx_labels - one might wonder where we have the two, rather than e.g. just extra_labels (I know: we discussed this, and I am not saying we should not have two parameters; but there should be some explanation as to _why_).

something like so:

creates or updates a github issue for the given `scanned_element`. If no issue exists, yet, it will be created, otherwise, it is checked that at most one matching issue exists; which will then be updated.

Issues are found by labels. Some of those are derived from the scanned_element. If given, ctx_labels are also included in search-query (i.e. an issue must have all of the given ctx_labels to be considered a match).

All of those labels + any given `extra_labels` are assigned to the issue, both in case it is created anew, or updated. Extra labels that are not ignored via the `preserve_labels_regexes` argument are dropped.